corehq/apps/case_migrations/views.py

```diff
@@ -47,27 +50,38 @@ def form_valid(self, form):
         return HttpResponseRedirect(self.page_url)
 
 
-def get_case_and_descendants(domain, case_id):
+def get_case_hierarchy_for_restore(case):
     from corehq.apps.reports.view_helpers import get_case_hierarchy
-    case = CaseAccessors(domain).get_case(case_id)
-    return [c for c in get_case_hierarchy(case, {})['case_list']
-            if not c.closed]
+    return [
+        c for c in get_case_hierarchy(case, {})['case_list']
+        if not c.closed
+    ]
 
 
-@mobile_auth
+@location_safe
+@mobile_auth_or_formplayer(require_user=False)
```

Comment:
i think this one is still going to fail auth because there's no couch_user associated with the request and `_login_or_challenge` requires it: https://github.com/dimagi/commcare-hq/blob/bbc6bb714be3bd5a512c24dd130d9c4c5e31033b/corehq/apps/domain/decorators.py#L209

Reply:
Completely correct. I was trying to keep session auth as an option since it makes it easy to test / check on a live server but I'll drop that for simplicity reasons. 

I only finished this late yesterday so hadn't had a chance to go though it again and when I initially tested it was working because I had the `anonymous user` flag on.